File: ts/packages/luzid-grpc-client/src/client/mutator.ts
import { createClient } from '../connection'
import type { Channel } from '../connection'
import {
  MutatorCloneAccountRequest,
  MutatorCloneAccountResponse,
  MutatorCloneAccountServiceDefinition,
  MutatorCloneAccountServiceClient,
  MutatorModifyAccountServiceClient,
  MutatorModifyAccountServiceDefinition,
  MutatorModifyAccountRequest,
  MutatorModifyAccountResponse,
} from '@luzid/grpc'

export {
  MutatorCloneAccountRequest,
  MutatorCloneAccountResponse,
} from '@luzid/grpc'

// -----------------
// MutatorCloneAccountClient
// -----------------
class MutatorCloneAccountClient {
  private readonly client: MutatorCloneAccountServiceClient

  constructor(channel: Channel) {
    this.client = createClient(MutatorCloneAccountServiceDefinition, channel)
  }

  cloneAccount(
    request: MutatorCloneAccountRequest
  ): Promise<MutatorCloneAccountResponse> {
    return this.client.cloneAccount(request)
  }
}

// -----------------
// MutatorModifyAccountClient
// -----------------
class MutatorModifyAccountClient {
  private readonly client: MutatorModifyAccountServiceClient

  constructor(channel: Channel) {
    this.client = createClient(MutatorModifyAccountServiceDefinition, channel)
  }

  modifyAccount(
    request: MutatorModifyAccountRequest
  ): Promise<MutatorModifyAccountResponse> {
    return this.client.modifyAccount(request)
  }
}

// -----------------
// Consolidated MutatorClient
// -----------------
export class MutatorClient {
  private readonly cloneAccountClient: MutatorCloneAccountClient
  private readonly modifyAccountClient: MutatorModifyAccountClient

  constructor(channel: Channel) {
    this.cloneAccountClient = new MutatorCloneAccountClient(channel)
    this.modifyAccountClient = new MutatorModifyAccountClient(channel)
  }

  cloneAccount(
    request: MutatorCloneAccountRequest
  ): Promise<MutatorCloneAccountResponse> {
    return this.cloneAccountClient.cloneAccount(request)
  }

  modifyAccount(
    request: MutatorModifyAccountRequest
  ): Promise<MutatorModifyAccountResponse> {
    return this.modifyAccountClient.modifyAccount(request)
  }
}

File: dart/luzid-grpc-client/lib/src/client/mutator.dart
import 'package:grpc/grpc_connection_interface.dart';
import 'package:luzid_grpc/luzid_grpc.dart';

// -----------------
// MutatorCloneAccountClient
// -----------------
class MutatorCloneAccountClient {
  final MutatorCloneAccountServiceClient _client;

  MutatorCloneAccountClient(ClientChannelBase channel)
      : _client = MutatorCloneAccountServiceClient(channel);

  Future<MutatorCloneAccountResponse> cloneAccount(
    MutatorCloneAccountRequest request,
  ) {
    return _client.cloneAccount(request);
  }
}

// -----------------
// MutatorModifyAccountClient
// -----------------
class MutatorModifyAccountClient {
  final MutatorModifyAccountServiceClient _client;

  MutatorModifyAccountClient(ClientChannelBase channel)
      : _client = MutatorModifyAccountServiceClient(channel);

  Future<MutatorModifyAccountResponse> modifyAccount(
    MutatorModifyAccountRequest request,
  ) {
    return _client.modifyAccount(request);
  }
}

// -----------------
// MutatorAccountClonedSubClient
// -----------------
class _MutatorAccountClonedSubClient {
  final MutatorAccountClonedSubClient _client;

  _MutatorAccountClonedSubClient(ClientChannelBase channel)
      : _client = MutatorAccountClonedSubClient(channel);

  ResponseStream<MutatorAccountCloned> subAccountCloned() {
    return _client.subMutatorAccountCloned(Empty());
  }
}

// -----------------
// Consolidated MutatorClient
// -----------------
class MutatorClient {
  final MutatorCloneAccountClient _cloneAccountClient;
  final MutatorModifyAccountClient _modifyAccountClient;
  final _MutatorAccountClonedSubClient _accountClonedSubClient;

  MutatorClient(ClientChannelBase channel)
      : _cloneAccountClient = MutatorCloneAccountClient(channel),
        _modifyAccountClient = MutatorModifyAccountClient(channel),
        _accountClonedSubClient = _MutatorAccountClonedSubClient(channel);

  Future<MutatorCloneAccountResponse> cloneAccount(
    MutatorCloneAccountRequest request,
  ) {
    return _cloneAccountClient.cloneAccount(request);
  }

  Future<MutatorModifyAccountResponse> modifyAccount(
    MutatorModifyAccountRequest request,
  ) {
    return _modifyAccountClient.modifyAccount(request);
  }

  ResponseStream<MutatorAccountCloned> subAccountCloned() {
    return _accountClonedSubClient.subAccountCloned();
  }
}
